### ISSUE TITLE
Seems like the identity for mat2d is set incorrectly?

### ISSUE THREAD

**Issue by baseten** [toji/gl-matrix#360]
It looks like the identity is being set incorrectly both on https://github.com/toji/gl-matrix/blob/master/src/mat2d.js#L36 and https://github.com/toji/gl-matrix/blob/master/src/mat2d.js#L84 

Correct me if I'm wrong but should it not be `out[3] = 0; out[4] = 1;`? It seems like `b` not `d` is being set to 1

**Comment by baseten** [comment]
Ah, sorry I see now this is a column/row major order thing. Is the doc bloc the culprit? https://github.com/toji/gl-matrix/blob/master/src/mat2d.js#L10

**Comment by stefnotch** [comment]
Yep, the documentation is apparently wrong. In the documentation of that class, the columns and rows have been switched.
https://github.com/toji/gl-matrix/blob/9ea87effc0ca32feb8511a01685ee596c6160fcc/src/mat2d.js#L107

Feel free to create a PR

**Comment by baseten** [comment]
Is this intentional as per note on [homepage](http://glmatrix.net/)? I understand wanting to follow OpenGL convention for column major examples, but since this appears to be a code block it's extra confusing 😄

**Comment by stefnotch** [comment]
The code is intentional, the mat2d documentation comment is wrong. A good example is [the translate function](https://github.com/toji/gl-matrix/blob/9ea87effc0ca32feb8511a01685ee596c6160fcc/src/mat2d.js#L239). As you can see, it affects `[4]` and `[5]`, which are `tx` and `ty` respectively. However, according to the wrong comment, `tx` is `[3]` and `ty` is `[5]`.

It's supposed to be
```
/**
 * 2x3 Matrix
 * @module mat2d
 *
 * @description
 * A mat2d contains six elements defined as:
 * <pre>
 * [a, b, c,
 *  d, tx, ty]
 * </pre>
 * This is a short form for the 3x3 matrix:
 * <pre>
 * [a, b, c,
 *  d, tx, ty,
 *  0, 0, 1]
 * </pre>
 * The last row is ignored so the array is shorter and operations are faster.
 */
```

**Comment by baseten** [comment]
OK, great, that's what I thought

**Comment by stefnotch** [comment]
Oh, yes! I made a mistake, was really careless when writing down the 3x3 matrix. Thanks for catching it.

**Comment by stefnotch** [comment]
Fixed by #361
